onmt/modules/copy_generator.py

```diff
@@ -177,7 +177,7 @@ def forward(self, scores, align, target):
         return loss
 
 
-class CommonCopyGeneratorLossCompute(CommonLossCompute):
+class CommonCopyGeneratorLossCompute(LossComputeBase):
```

Comment:
I'm not sure to grasp the whole rationale behind the `CommonLossCompute`/`LossComputeBase` refactoring. Is the last big remaining difference only the log_ppl computation?

Reply:
(Underlying question is: do we really need both `CommonLossCompute` and `LossComputeBase` anymore?)

Reply:
The _compute_loss, _make_shard_state (I need to move it back to the CommonLossCompute) and the way to use the generator are different between CopyGeneratorLoss and the other classes

Reply:
We can do it in one class, the code is already not very clear, it's not going to be worse. If we do that CopyGenerator will override _compute_loss, _compute_log_ppl and _compute_alignement_loss will only be used in the compute_loss of the main class

Reply:
> We can do it in one class, the code is already not very clear, it's not going to be worse. If we do that CopyGenerator will override _compute_loss, _compute_log_ppl and _compute_alignement_loss will only be used in the compute_loss of the main class

Yes I think this might be a bit better to explicitly override this method instead of having a full class that we don't really know what it's for unless we look at this specific CopyGeneratorLoss.

Reply:
I merged it, the ppl part is not nice. Also there is a normalization args that was not used anywhere, I will investigate to see if the normalization process disappeared by mistake

Reply:
normalization was already not used a year ago https://github.com/OpenNMT/OpenNMT-py/blob/78351302b0761178581d92612b528f6eea529618/onmt/utils/loss.py